apps/server/src/controllers/userControllers/authUsers.ts

```diff
@@ -247,10 +247,14 @@ export const forgotPassword = async (req: Request, res: Response) => {
         const longString = generateLongString(80) //generate 80 chars long random string
 
         // create unique verify token
-        const token = await Jwt.sign({
-          otp: otp.toString(),
-          tokenExpiry: tokenExpiry.getTime().toString(),
-        })
+        const token = await Jwt.sign(
+          {
+            id: user.id,
+            otp: otp.toString(),
+            tokenExpiry: tokenExpiry.getTime().toString(),
+          },
+          { _secret: Env.JWT_SECRET + longString }
```

Comment:
if you do this, how do you intend verifying the token?

Reply:
ohh I see, from the link?

hmmm... is it safe to leave it this way?

I still recommend patching up some random characters


you could create a new secret and put it in .env or slice from existing secrets

Reply:
or

![image](https://github.com/Ajo-Savings-Project/ajo-savings/assets/48376769/21673693-dfc5-4c0a-8ca5-fd71ed33fa26)


This way, we generate secret and long string id
when it's time to validate
We use the long string to pull out the data
then from that data, we get the secret
we then append the secret to the JWT existing access token secret
use that to validate the JWT, if the signature is valid, we proceed